packages/flutter/lib/src/services/text_input.dart

```diff
@@ -633,6 +633,9 @@ abstract class TextInputClient {
 
   /// Updates the floating cursor position and state.
   void updateFloatingCursor(RawFloatingCursorPoint point);
+
+  /// Connection close request is received.
+  void connectionClosed();
```

Comment:
Is this a request to close the connection (and the framework responds with actually closing the connection)? Or is this to inform the framework that the connection has been closed already (and the framework should not respond with closing it again)?

Reply:
Right now its name suggests the latter, but the responds is currently the former, I think?

Reply:
Shall we use a method name like "ConnectionCloRequest"? 

@mdebbar do you have a suggestion?

Reply:
I personally prefer the latter (the platform message informs the framework that the connection has been closed, and the framework doesn't respond).

For the web, both approaches are possible, I believe. Not sure about Android and iOS.

Reply:
Thanks I'll continue with the latter(connectionClosed), then.

Reply:
The current implementation calls _closeInputConnectionIfNeeded, though. Which will make the frame respond by closing the connection?

Also: We need to double-check that this model would work on other platforms like iOS and Android.

Reply:
The implementation of _closeInputConnectionIfNeeded:
void _closeInputConnectionIfNeeded() {
    if (_hasInputConnection) {
      _textInputConnection.close();
      _textInputConnection = null;
      _lastKnownRemoteTextEditingValue = null;
    }
  }

I believe this implementation is correct and this is what we need to do for closing the connection after the platform message is received. 

Shall I rename this method instead? Am I missing anything?

Reply:
`_textInputConnection.close();` sends a platform message (`TextInput.clearClient`) back to the engine asking it to close the connection. But since the engine is the one that requested the connection to be closed, there's no need for the framework to send the `TextInput.clearClient` message.

Reply:
done. Also updated the comments.